### DISCUSSION TITLE
MDAnalysis.analysis.dihedrals.Janin

### DISCUSSION THREAD

**Discussion by H-EKE** [MDAnalysis/mdanalysis#4567] [Q&A] [answered]
Hi,

I am trying to calculate chi1,chi2 tryptophan angles for a linked and unlinked system. I would like to use the Janin module to compute it. As my two systems are different (absence/presence linker), should i create a reference and align trajectories to this one ? or fit the whole trajectory to my tryptophan? 

At the end I would like to map identical tryptophan rotamers in the 2 systems.
At the end with my analysis, I would expect that similar regions would have the same rotamer. 
<img width="419" alt="exemple2" src="https://github.com/MDAnalysis/mdanalysis/assets/121638941/eda058ca-1ba4-4258-aa30-21103235e6d3">

Thanks in advance!

**Comment by orbeckst** [comment, marked as answer]
For any internal coordinates you do not have to do any trajectory processing. MDAnalysis knows how to take periodic boundaries into account and there's also nothing to be gained from superimposing. Just calculate the dihedral angles for your two trajectories and compare just as in your sketch.

**Reply by H-EKE** [reply]
Dear Oliver,

Thanks a lot for your answer! It helped me a lot!!

**Comment by H-EKE** [comment]
Just a last question,

Would it be a way more efficient to map frames regions with similar rotamers than this?
```
def frame_similarity(frame1, frame2, threshold):
    # Compare the absolute difference between corresponding angles
    differences = np.abs(np.array(frame1) - np.array(frame2))
    # If all the differences are below the threshold, consider the frames similar
    return np.all(differences < threshold)

# Define a threshold for similarity
similarity_threshold = 60 
# Find overlapping frames based on similarity
overlapping_frames = []
for idx, (angles, angles1) in enumerate(zip(janin.angles, janin1.angles)):
    if frame_similarity(angles, angles1, similarity_threshold):
        overlapping_frames.append(idx)

print("Overlapping Frames:", overlapping_frames)
print("Number of overlapping frames:", len(overlapping_frames))
```

**Reply by hmacdope** [reply]
Your approach seems valid, binning them using some similarity statistic, but your question is not well framed. 

"Would it be a way more efficient to map frames regions with similar rotamers than this?" Is unclear. Would you be able to clarify what you mean?
